components/src/jsMain/kotlin/dev/fritz2/components/file.kt

```diff
@@ -0,0 +1,252 @@
+            }
+        }
+    }
+
+    var fileReadingStrategy: FileReadingStrategy = base64
+
+    fun encoding(value: String) {
+        fileReadingStrategy = plainText(value)
+    }
+
+    internal var renderContext: RenderContext.(HTMLInputElement) -> Unit = { input ->
+        pushButton(prefix = "file-button") {
+            icon { fromTheme { cloudUpload } }
+            element {
+                domNode.onclick = {
+                    input.click()
+                }
+            }
+        }
+    }
+
+    open fun button(
+        styling: BasicParams.() -> Unit = {},
+        baseClass: StyleClass? = null,
+        id: String? = null,
+        prefix: String = "file-button",
+        build: PushButtonComponent.() -> Unit = {}
+    ) {
+        renderContext = { input ->
+            pushButton(styling, baseClass, id, prefix) {
+                build()
+                element {
+                    domNode.onclick = {
+                        input.click()
+                    }
+                }
+            }
+        }
+    }
+}
+
+/**
+ * This component generates a single file selection context.
+ *
+ * In there you can create a button with a label, an icon, the position of the icon and access its events.
+ * For a detailed overview about the possible properties of the button component object itself, have a look at
+ * [PushButtonComponent]
+ *
+ * The [File] function then returns a [Flow] of [File] in order
+ * to combine the [Flow] directly to a fitting _handler_ which accepts a [File]:
+ * ```
+ * val textFileStore = object : RootStore<String>("") {
+ *     val upload = handle<File> { _, file -> file.content }
+ * }
+ * file {
+ *   accept("text/plain")
+ *   encoding("utf-8")
+ *   button(id = "myFile") {
+ *     text("Select a file")
+ *   }
+ * } handledBy textFileStore.upload
+ * ```
+ *
+ * @see PushButtonComponent
+ *
+ * @param styling a lambda expression for declaring the styling as fritz2's styling DSL
+ * @param baseClass optional CSS class that should be applied to the element
+ * @param id the ID of the element
+ * @param prefix the prefix for the generated CSS class resulting in the form ``$prefix-$hash``
+ * @param build a lambda expression for setting up the component itself. Details in [PushButtonComponent]
+ * @return a [Flow] that offers the selected [File]
+ */
+fun RenderContext.file(
+    styling: BasicParams.() -> Unit = {},
+    baseClass: StyleClass? = null,
+    id: String? = null,
+    prefix: String = "file",
+    build: FileSelectionComponent.() -> Unit = {}
+): Flow<File> {
+    var file: Flow<File>? = null
+    val component = FileSelectionComponent().apply(build)
+    (::div.styled(styling, baseClass, id, prefix) {}) {
+        val inputElement = (::input.styled { display { none } }) {
+            type("file")
+            component.accept?.invoke(this)
+            file = changes.events.mapNotNull {
+                domNode.files?.item(0)
```

Comment:
Might still let file to be null right? 
If so, the return in line 186 could fail!

Could this also be done without a nullable var? Or with a ``Flow<File?>`` instead?